BUILD.md

```diff
@@ -0,0 +1,173 @@
+# Building the Arkouda Server
+
+## Table of Contents
+1. [Getting Started](#start)
+2. [Environment Variables](#env-vars)
+3. [Building the Source](#build-ak-source)
+   1. [Using Environment Installed Dependencies](#env_deps)
```

Comment:
should we mark `Using Environment Installed Dependencies` as the recommended

Reply:
Probably a good idea. Updated with `(Recommended)`